avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1567,9 +1593,7 @@ private Schema toConnectSchema(org.apache.avro.Schema schema,
       case ARRAY:
         org.apache.avro.Schema elemSchema = schema.getElementType();
         // Special case for custom encoding of non-string maps as list of key-value records
-        if (elemSchema.getType().equals(org.apache.avro.Schema.Type.RECORD)
-            && NAMESPACE.equals(elemSchema.getNamespace())
-            && MAP_ENTRY_TYPE_NAME.equals(elemSchema.getName())) {
+        if (isMapEntry(elemSchema)) {
```

Comment:
shouldn't we restore the `org.apache.avro.Schema.name()` somewhere here?

Reply:
It's already handled - fromConnectSchema puts the name in the "connect.name" property, and toConnectSchema restores it from there.